src/lib/DefaultRuleManager.ts

```diff
@@ -40,27 +49,47 @@ export class DefaultRuleManager implements RuleManager {
 		return this.catalog.getRulesMatchingFilters(filters);
 	}
 
-	async runRulesMatchingCriteria(filters: RuleFilter[], target: string[] | string, format: OUTPUT_FORMAT): Promise<string> {
-		// If target is a string, it means it's an alias for an org, instead of a bunch of local file paths. We can't handle
-		// running rules against an org yet, so we'll just throw an error for now.
-		if (typeof target === 'string') {
-			throw new SfdxError('Running rules against orgs is not yet supported');
-		}
+	async runRulesMatchingCriteria(filters: RuleFilter[], targets: string[], format: OUTPUT_FORMAT): Promise<string | { columns; rows }> {
+		let results: RuleResult[] = [];
 
-		// Convert our filters into paths that we can feed to the engines. If we didn't find any paths, we're done.
-		const paths: PathGroup[] = await this.catalog.getNamedPathsMatchingFilters(filters);
-		if (paths == null || paths.length === 0) {
-			this.logger.trace('No Rule paths found. Nothing to execute.');
-			return '';
+		// Derives rules from our filters to feed the engines.
+		const ruleGroups: RuleGroup[] = await this.catalog.getRuleGroupsMatchingFilters(filters);
+		const rules: Rule[] = await this.catalog.getRulesMatchingFilters(filters);
+		const ps: Promise<RuleResult[]>[] = [];
+		for (const e of this.engines) {
+			const engineGroups = ruleGroups.filter(g => g.engine === e.getName());
+			const engineRules = rules.filter(r => r.engine === e.getName());
+			const engineTargets = await this.unpackTargets(e, targets);
+			ps.push(e.run(engineGroups, engineRules, engineTargets));
 		}
-
-		const ps = this.engines.map(e => e.run(paths, target));
-		const [results] = await Promise.all(ps);
+		const psResults: RuleResult[][] = await Promise.all(ps);
+		psResults.forEach(r => results = results.concat(r));
 		this.logger.trace(`Received rule violations: ${results}`);
-
-		// Once all of the rules finish running, we'll need to combine their results into a single set of the desired type,
-		// which we can then return.
 		this.logger.trace(`Recombining results into requested format ${format}`);
-		return RuleResultRecombinator.recombineAndReformatResults([results], format);
+		return RuleResultRecombinator.recombineAndReformatResults(results, format);
+	}
+
+	private async unpackTargets(engine: RuleEngine, targets: string[]): Promise<string[]> {
+		// Add any default target patterns from config.
```

Comment:
Would this be patterns that are always targeted regardless of input? Do you have any examples?

Reply:
Yep. This is how we prevent eslint from trying to scan .cls files, for example.  And how we exclude node_modules by default.  

We do need that other story about supporting a local config.json in the current working project directory, which can override or specify its own target patterns.

Reply:
I think it would helpful to have a comment explaining that.

Reply:
Yeah, I added a bunch of commentary about it per your other comment.